python/src/FieldFunctionalChaosSobolIndices_doc.i.in

```diff
@@ -0,0 +1,153 @@
+%feature("docstring") OT::FieldFunctionalChaosSobolIndices
+"Sobol indices from a functional decomposition.
+
+This class allows to perform sensitivity analysis from field chaos decomposition.
+The process decomposition is done by Karhunen-Loeve and the modes interpolation
+is performed by functional chaos:
+
+.. math::
+
+    \tilde{g}(x) = \sum_{k \in K_c} \beta_{\vect{\alpha}_k} \Psi_{\vect{\alpha}_k}(x)
+
+Lets expand the multi indices notation:
+
+.. math::
+
+    \Psi_{\vect{\alpha}}(x) = \prod_{j=1}^{K_T} P^j_{\alpha_j}(x_j)
+
+with
+
+.. math::
+
+    \vect{\alpha} \in \mathbb{N}^{K_T} = \{\underbrace{\alpha_1, \dots, \alpha_{K_1}}_{K_1},\dots,\underbrace{\alpha_{K_T-K_d}, \dots, \alpha_{K_T}}_{K_d}\}
+
+Sobol indices of the input field component :math:`j \in [1,d]` can be computed
+from the coefficients of the chaos decomposition that involve the
+matching Karhunen-Loeve coefficients.
+
+For the first order Sobol indices we sum over the multi-indices :math:`\vect{\alpha}_k`
+that are non-zero on the :math:`K_j` indices corresponding to the Karhunen-Loeve
+decomposition of j-th input and zero on the other :math:`K_T - K_j` indices (noted :math:`G_j`):
+
+.. math::
+
+    S_j = \frac{\sum_{k=1, \vect{\alpha}_k \in G_j}^{K_c} \beta_{\vect{\alpha}_k}^2}{\sum_{k=1}^{K_c} \beta_{\vect{\alpha}_k}^2}
+
+For the total order Sobol indices we sum over the multi-indices :math:`\vect{\alpha}_k`
+that are non-zero on the :math:`K_j` indices corresponding to the Karhunen-Loeve
+decomposition of the j-th input (noted :math:`GT_j`):
+
+.. math::
+
+    S_{T_j} = \frac{\sum_{k=1, \vect{\alpha}_k \in GT_j}^{K_c} \beta_{\vect{\alpha}_k}^2}{\sum_{k=1}^{K_c} \beta_{\vect{\alpha}_k}^2} 
+
+This generalizes to higher order indices.
+
+Parameters
+----------
+result : :class:`~openturns.FieldFunctionalChaosResult`
+    Result.
+
+See also
+--------
+FieldToPointFunctionalChaosAlgorithm
+
+Examples
+--------
+>>> import openturns as ot
+>>> ot.RandomGenerator.SetSeed(0)
+>>> mesh = ot.RegularGrid(0.0, 0.1, 20)
+>>> cov = ot.KroneckerCovarianceModel(ot.MaternModel([2.0], 1.5), ot.CovarianceMatrix(4))
+>>> X = ot.GaussianProcess(cov, mesh)
+>>> x = X.getSample(500)
+>>> y = [[m[0] + m[1] + m[2] - m[3] + m[0] * m[1] - m[2] * m[3] - 0.1 * m[0] * m[1] * m[2]] for m in [xi.computeMean() for xi in x]]
+>>> algo = ot.FieldToPointFunctionalChaosAlgorithm(x, y)
+>>> algo.setThreshold(4e-2)
+>>> bs = ot.ResourceMap.GetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize')
+>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', 100)
+>>> algo.run()
+>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', bs)
```

Comment:
This business with the resource map key should be explained as a comment in the code, I think. Otherwise users might wonder why you are setting something only to unset it 2 lines below.

```suggestion
>>> # Temporarily lower the basis size for the sake of this example.
>>> # We need to store the original size.
>>> bs = ot.ResourceMap.GetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize')
>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', 100)
>>> algo.run()
>>> # The algorithm has been run with the lower basis size:
>>> # we can now restore the original value.
>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', bs)
```